overlord/snapstate/snapmgr.go

```diff
@@ -659,10 +674,15 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// store for undo
+	t.Set("old-current", snapst.Current)
```

Comment:
I would follow the style of the other old- in here, so set oldCurrent just before modifying snapst.Current, and storing it in the same bunch as the other old-